### PR TITLE
[Merged by Bors] - feat(Algebra/Module/Basic): add RingHom.smulOneHom

### PR DESCRIPTION
**Pull request by eric-wieser**
This also renames the existing `smulOneHom` to `MonoidHom.smulOneHom`.

Co-authored-by: Junyan Xu <junyanxu.math@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Defs.lean [hunk 1]
```diff
@@ -705,15 +705,15 @@ theorem SMulCommClass.of_mul_smul_one {M N} [Monoid N] [SMul M N]
...
+def MonoidHom.smulOneHom {M N} [Monoid M] [MulOneClass N] [MulAction M N] [IsScalarTower M N N] :
     M →* N where
   toFun x := x • (1 : N)
   map_one' := one_smul _ _
-  map_mul' x y := by rw [smul_one_mul, smul_smul]
-#align smul_one_hom smulOneHom
-#align vadd_zero_hom vaddZeroHom
-#align smul_one_hom_apply smulOneHom_apply
-#align vadd_zero_hom_apply vaddZeroHom_apply
+  map_mul' x y := by dsimp; rw [smul_one_mul, smul_smul]
```
Comment (alreadydone) [inline Mathlib/GroupTheory/GroupAction/Defs.lean:712]: ?
```suggestion
  map_mul' x y := by rw [smul_one_mul, smul_smul]
```

**Review by alreadydone — APPROVED** [review]
Thanks, LGTM!

cc author of `mul_one_hom` @urkud in https://github.com/leanprover-community/mathlib/pull/17058

**Comment by alreadydone** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**